### PR TITLE
Add missiles to the Journal. Needs to be adjusted to 0.7.1 yet

### PR DESCRIPTION
**Pull request by spixi**
Because of the Huntress redo, missiles are now much more important. This adds a missile menu to the Journal.

### CODE REVIEW

File: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/items/weapon/missiles/MissileWeapon.java [hunk 1]
```diff
@@ -44,9 +44,13 @@
...
 abstract public class MissileWeapon extends Weapon {
 
+	protected MissileWeapon( ) {
+		hitsToKnow = 2;
+	}
+
 	{
 		stackable = true;
-		levelKnown = true;
+		levelKnown = false;
```
Comment (spixi) [inline core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/items/weapon/missiles/MissileWeapon.java]: Items are currently only identified, if their level was unknown before. That makes it also a surprise to find secret +1 or even +2 missiles 0.7.1+ as a side effect.

File: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/hero/HeroClass.java [hunk 2]
```diff
@@ -155,7 +155,7 @@ private static void initRogue( Hero hero ) {
 		hero.belongings.misc1.activate( hero );
 
 		ThrowingKnife knives = new ThrowingKnife();
-		knives.quantity(3).collect();
+		knives.identify().quantity(3).collect();
```
Comment (spixi) [inline core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/hero/HeroClass.java]: This was somehow forgotten.

File: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/items/weapon/missiles/darts/TippedDart.java [hunk 3]
```diff
@@ -61,6 +61,7 @@ public int STRReq(int lvl) {
 	protected void rangedHit(Char enemy, int cell) {
 		//attempt to stick the dart to the enemy, just drop it if we can't.
 		Dart d = new Dart();
+		d.identify();
```
Comment (spixi) [inline core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/items/weapon/missiles/darts/TippedDart.java]: It's not a mystery to identify the Dart to the player.

**Comment by spixi** [comment]
screenshot_2018-12-16-04-09-32-756_com shatteredpixel shatteredpixeldungeon indev

**Comment by 00-Evan** [comment]
Thank you for the effort, but I'm afraid I don't accept pull requests. Shattered is a solo project and the source is made available for others to build from, not to allow contributions.

Regarding the functionality itself, I do agree that there are some items which probably should be cataloged, but the issue for the moment is that the existing catalog system relies on items being identifiable, and thrown weapons are not. I don't have any plans to make them identifiable in 0.7.1

**Pull request closed without merging**